integration-tests/jt400/pom.xml

```diff
@@ -63,6 +71,9 @@
             </activation>
             <properties>
                 <quarkus.package.type>native</quarkus.package.type>
+                <!-- Flat class has to be configured in build-time for native mode.
+                  If property is not set, the native test will fail because build-time property is changed in runtime via testProfile -->
+                <quarkus.test.flat-class-path>${flat-class-path}</quarkus.test.flat-class-path>
```

Comment:
you tip would work correctly. I tried it before. I added Profile as a verification.
The real behavior in native is, that the native succeeds in case of flat-class-path = true or false. (TBH I'm not entirely sure why this is happening)
The profile changes flat class pat in runtime and brings 2 options:

1. if native run starts with flat-class-path = false, and profile changes value to true, the execution fails. Quarkus detects, that build time property is changed in runtime
2. It should be possible to run the mock test in JVM wholy based on profile. (just the fact)

Therefore I kept profile to make sure that native value is the same as JVM value (even if it has no effect on the result of the test)

I can try to simplify a little bit by ignoring  `flat-class-path` and using only property `quarkus.test.flat-class-path`. I'd like too keep the profile just as another level of  verification (to avoid false positive)

Reply:
> The real behavior in native is, that the native succeeds in case of flat-class-path = true or false. TBH I'm not entirely sure why this is happening)

Because the application and tests are running in separate processes. There are some differences in class loading in test mode Vs prod mode. See the end of the introductory paragraph here:

https://quarkus.io/guides/class-loading-reference

Reply:
~~I understand that fact, but the mock backend is part of the application as well as the jtopen library, which is extended using flat-classpath~~

I see the last part of the paragraph, therefore it has probably no meaning to keep tee profile as verification, layer because native "ignores" `flat-class-path`. I'll simplify the code as much as possible.

Thanks for the link!

Reply:
> Maybe a better way of doing this is to just set `quarkus.test.flat-class-path = true` in `application.properties`. Then change the `skip-mock-tests` profile to include property `<quarkus.test.flat-class-path>false</quarkus.test.flat-class-path>`.
> 
> Then there's no need for the `flat-class-path` Maven property or `FlatClasspathTestProfile.java`.
> 
> I may be wrong, but I think it could work.

I tried the approach you described (application properties with `quarkus.test.flat-class-path=true` and profile with property `<quarkus.test.flat-class-path>false</quarkus.test.flat-class-path>`. Flat class-path is true even when profile is activated
.
I just remembered that I had to add resource filtering to override the application.property.
I'll do it in that way, I think that it is still much simpler then my original approach.